draft-trammell-taps-interface.md

```diff
@@ -1243,28 +1024,93 @@ connectionProperties := Connection.getProperties()
 
 {{appendix-specify-query-params}} gives a more detailed overview of the different types of properties that can be set and queried at different times.
 
+## Protocol Properties {#protocol-props}
+
+Protocol Properties represent the configuration of the selected Protocol Stacks
+backing a connection. Some properties apply generically across multiple
+transport protocols, while other properties only apply to specific protocols.
+The default settings of these properties will vary based on the specific
+protocols being used and the system's configuration.
+
+Note that Protocol Properties are also set during pre-establishment, as
+transport parameters, to preconfigure Protocol Stacks during establishment.
+
+Generic Protocol Properties include:
+
+* Timeout for aborting Connection: 
+  This numeric property specifies how long
+  to wait before aborting a Connection during establishment, or after a
+  connection has failed after establishment. It is given in seconds.
+
+* Abort timeout to suggest to the Remote Endpoint: 
+  This numeric property
+  specifies the timeout to propose to the Remote Endpoint. It is given in
+  seconds.
+
+* Retransmission threshold before excessive retransmission notification:
+  This numeric property specifies after how many retransmissions to inform
+  the application about "Excessive Retransmissions".
+
+* Required minimum coverage of the checksum for receiving:
+  This numeric property specifies the part of the received data that needs
+  to be covered by a checksum. It is given in Bytes. A value of 0 means
+  that no checksum is required, and a special value (e.g., -1) indicates
+  full checksum coverage.
+
+* Connection group transmission scheduler: 
+  This enumerated property specifies which
+  scheduler should be used among Connections within a Connection Group. It
+  applies to connection groups; the set of schedulers can be taken from
+  {{I-D.ietf-tsvwg-sctp-ndata}}.
+
+* Maximum message size concurrent with connection establishment:
+  This numeric property represents the maximum Message size that can be sent
+  before or during Connection establishment, see also {{send-idempotent}}.
+  It is given in Bytes. This property is read-only.
+
+* Maximum Message size before fragmentation or segmentation:
+  This numeric property, if applicable, represents the maximum Message size 
+  that can be sent without incurring network-layer fragmentation and/or transport layer segmentation at the sender. This property is read-only.
+
+* Maximum non-partial Message size on send:
+  This numeric property represents the maximum Message size that can be sent as
+  a non-partial Message. This property may be read-only.
+
+* Maximum non-partial Message size on receive: 
+  This numeric property represents
+  the maximum Message size that can be received as a non-partial Message. This property may be read-only.
```

Comment:
Why "may be"?  The ones above just read "...is read-only", which seems better.

Reply:
Same rationale as above - a function of the transport protocol, imposing a max value.